solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -38,6 +38,15 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
         uint256 timestamp;          // timestamp
     }
 
+    struct PackedRates {
+        uint128 addSpotRateN;
```

Comment:
did you consider using uint128 for fractions?

Reply:
We can't do that of course, because we're using `uint256` based Fractions everywhere.

Reply:
the Fraction struct is only used inside contracts, not between contracts and also not in storage.
we could choose to use 128 bit based fraction in the liquidity protection which would have saved us some headaches, but only if it would actually work, since I know also multiply them by one another etc.
so not sure if it's practical but just a suggestion.